src/metadata/taglib_handler.cc

```diff
@@ -416,12 +416,16 @@ void TagLibHandler::extractMP3(TagLib::IOStream* roStream, const std::shared_ptr
             if (frameList.isEmpty())
                 continue;
 
-            const TagLib::ID3v2::Frame* frame = frameList.front();
-            const auto textFrame = dynamic_cast<const TagLib::ID3v2::TextIdentificationFrame*>(frame);
-
-            const TagLib::String frameContents = textFrame->toString();
-            std::string value(frameContents.toCString(true));
-            value = sc->convert(value);
+            std::string value;
+            for (const auto& frame : frameList) {
+                const auto textFrame = dynamic_cast<const TagLib::ID3v2::TextIdentificationFrame*>(frame);
```

Comment:
You need to check that `dynamic_cast` succeed. Even if works now who knows about future versions...